### PR TITLE
Cleanup of logs page

### PR DESCRIPTION
**Pull request by cheld**
- Renamed a lot
- Structed the result object in 3 parts (ui-part, selection slice, and logs)
- Remove duplicated selection object

### CODE REVIEW

File: src/app/externs/backendapi.js [hunk 1]
```diff
@@ -918,12 +917,20 @@ backendApi.LogLine;
 
 /**
  * @typedef {{
- *   referenceLogLineId: !backendApi.LogLineReference,
- *   relativeFrom: number,
- *   relativeTo: number
+ *   referencePoint: !backendApi.LogLineReference,
```
Comment (floreks) [inline src/app/externs/backendapi.js]: Isn't type of this `LogPageInfo`?

File: src/app/backend/resource/logs/logs.go [hunk 2]
```diff
@@ -46,60 +45,71 @@ var OldestLogLineId = LogLineId{
...
-type LogViewSelector struct {
-	// ReferenceLogLineId is the ID of a line which should serve as a reference point for this selector.
-	// You can set it to last or first line if needed. Setting to the first line will result in standard slicing.
-	ReferenceLogLineId LogLineId
-	// First index of the slice relatively to the reference line(this one will be included).
-	RelativeFrom int
-	// Last index of the slice relatively to the reference line (this one will not be included).
-	RelativeTo int
+// Representation of one page of logs
+type LogPage struct {
```
Comment (floreks) [inline src/app/backend/resource/logs/logs.go]: I'm struggling a bit with using `Page` as part of backend api structure. For me it suggests that it is in some correlation with web page that shows logs, however it is only part of our API and it is returned as JSON object.

WDYT about something like `LogChunk` `LogSlice` `LogPart`, something that suggests that it is not full log but also doesn't suggest that it has to be related to web view.

File: src/app/externs/backendapi.js (hunk 1 above)
Comment (cheld) [inline src/app/externs/backendapi.js]: The current structure is:

```
LogPage
      LogPageInfo (podname, containername, timestamps )
      LogSelection (reference point to keep track of paging)
      LogLines  (actual logs)
```
LogViewInfo -> LogSelection is syntactically correct. Still confusing?

File: src/app/backend/resource/logs/logs.go (hunk 2 above)
Comment (cheld) [inline src/app/backend/resource/logs/logs.go]: Ok, I am thinking about it

File: src/app/externs/backendapi.js (hunk 1 above)
Comment (floreks) [inline src/app/externs/backendapi.js]: Ok, It makes sense now. LogSelection name works for me.

File: src/app/backend/resource/logs/logs.go (hunk 2 above)
Comment (cheld) [inline src/app/backend/resource/logs/logs.go]: I was thinking about trying to make it consistent with other resources. 

Typically we name things *list or *details (podlist, poddetails). I was wondering if the following structure would make sense (or lets say indicates how it is to be used):

```
LogDetails {
     LogInfo (LogMeta?)
     LogSelection
     LogLines
} 
```

Comment (floreks) [inline src/app/backend/resource/logs/logs.go]: Sounds good to me.

**Comment by cheld** [comment]
PTAL

**Comment by cheld** [comment]
@floreks can I merge?

**Pull request merged**